sktime/forecasting/ensemble_algorithms.py

```diff
@@ -0,0 +1,234 @@
+from sklearn.base import BaseEstimator
+from scipy.optimize import nnls
+import numpy as np
+
+
+class EnsembleAlgorithm(BaseEstimator):
+    """Wrapper class to take ensemble algorithms and allow them to be easily set
+
+    Parameters
+    ----------
+    n : float
+    loss_func : function
+    """
+    def __init__(self, n=10, loss_func=None, **kwargs):
+        self.n = n
+        self.weights = np.ones(n)/n
+        self.loss_func = loss_func
+
+    def _predict(self, expert_predictions):
+        """ Performs prediction by taking a weighted average of the expert
+            predictions w.r.t the weights vector
+
+        Parameters
+        ----------
+        expert_predictions : np.array(), shape=(time_axis,experts_axis)
+
+        Returns
+        -------
+        predictions : np.array(), shape=(time_axis)
+        """
+        prediction = np.dot(self.weights, expert_predictions)
+
+        return prediction
+
+    def _modify_weights(self, new_array):
+        """Performs a pointwise multiplication of the current
+        weights with a new array of weights.
+
+        Parameters
+        ----------
+        new_array : np.array()
+        """
+        self.weights = self.weights * new_array
+        self.weights /= np.sum(self.weights)
+
+    def _update(self, expert_predictions, actual_values):
+        """ Resets the weights over the experts by passing previous observations
+            to the online_expert algorithm
+
+        Parameters
+        ----------
+        expert_predictions : np.array(), shape=(time_axis,experts_axis)
+        actual_values : np.array(), shape=(time_axis)
+        """
+        raise NotImplementedError()
+
+    def _uniform_weights(self, n):
+        """ Resets weights for n expert to uniform weights
+
+        Parameters
+        ----------
+        n : int
+        """
+        self.n = n
+        self.weights = np.ones(n)/n
+
+
+class HedgeExpertEnsemble(EnsembleAlgorithm):
+    """Wrapper class to take hedge style ensemble algorithms and
+       allow them to be updated
+
+    Parameters
+    ----------
+    n : float, number of experts
```

Comment:
please choose more meaningful variable names, e.g. `n_estimators` , here and throughout

Reply:
Certainly